pennylane/ops/op_math/exp.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines the symbolic operation that stands for an operator raised to a power.
+"""
+from warnings import warn
+from scipy.sparse.linalg import expm as sparse_expm
+
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import (
+    expand_matrix,
+    Tensor,
+    OperatorPropertyUndefined,
+)
+from pennylane.wires import Wires
+
+from .symbolicop import SymbolicOp
+
+
+def exp(op, coeff=1, id=None):
+    """Take the exponential of an Operator times a coefficient.
+
+    Args:
+        base (~.operation.Operator): The Operator to be exponentiated
+        coeff=1 (Number): A scalar coefficient of the operator.
```

Comment:
Should the type for coefficient be `Complex` in general?

Reply:
`Number` includes `Complex`. I think? Python isn't very good at typing.